deploy/template.yaml

```diff
@@ -1125,6 +1137,175 @@ Resources:
       DatapointsToAlarm: 1
       Threshold: 1
       ComparisonOperator: GreaterThanOrEqualToThreshold
+  
+  HmrcTokenFunction:
+    Type: AWS::Serverless::Function
+    DependsOn:
+      - "HmrcTokenLogGroup"
+    Properties:
+      FunctionName: !Sub "HmrcToken-${AWS::StackName}"
+      CodeUri: ../src/
+      Handler: HmrcTokenHandler.lambdaHandler
+      ReservedConcurrentExecutions: !If
+        - ApplyReservedConcurrency
+        - !Ref LambdaConcurrency
+        - !Ref AWS::NoValue
+      # Events:        
+      #   InvocationLevel:
+      #     Type: Schedule
+      #     Properties:
+      #       Schedule: cron(0/5 * * * ? *)          
+      Environment:
+        Variables:
+          HMRC_CLIENT_ID_SSM_PATH: !Sub "/${Environment}/HMRC/CLIENT_ID"
+          HMRC_CLIENT_SECRET_SSM_PATH: !Sub "/${Environment}/HMRC/CLIENT_SECRET"
+          HMRC_BASE_URL: !FindInMap [ EnvironmentVariables, !Ref Environment, HMRCBASEURL ]
+      Policies:
+        - AWSLambdaBasicExecutionRole
+        - AWSXrayWriteOnlyAccess
+        - SSMParameterReadPolicy:
+            ParameterName: !Sub "${Environment}/HMRC/CLIENT_ID"
+        - SSMParameterReadPolicy:
+            ParameterName: !Sub "${Environment}/HMRC/CLIENT_SECRET"
+        - Statement:
+            - Sid: SSMParameterWritePolicy
+              Effect: Allow
+              Action:
+                - ssm:PutParameter
+              Resource:
+                - !Sub "arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${Environment}/HMRC/TOKEN"
+    Metadata:
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: "es2020"
+        Sourcemap: true
+        EntryPoints:
+          - HmrcTokenHandler.ts
+  
+  HmrcTokenScheduledRule: 
+    Type: AWS::Events::Rule
+    Condition: NotLocalTestStack
+    Properties: 
+      Description: "ScheduledRule"
+      ScheduleExpression: cron(0/5 * * * ? *)
```

Comment:
I have added the cron to run every 5 minutes for now to test main stack deployment. Once its working, will update it to cron(0 0/3 * * ? *) and remove the commented lines. Currently main stack lambda is trying to execute but failing for PutParameter due to permissionsBoundary error.